cpp/src/arrow/array/concatenate.cc

```diff
@@ -436,6 +438,23 @@ class ConcatenateImpl {
     return Status::OK();
   }
 
+  Status Visit(const RunEndEncodedType& type) {
+    int64_t physical_length = 0;
+    for (auto input : in_) {
+      physical_length =
+          SafeSignedAdd(physical_length, ree_util::FindPhysicalOffset(ArraySpan(*input)));
```

Comment:
Are we sure this cannot overflow? If it can, we should detect overflow instead of letting it wrap around.